app/src/main/java/com/example/recipeapp/Fragments/FeedFragment.java

```diff
@@ -63,6 +65,7 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         // Inflate the layout for this fragment
         binding = FragmentFeedBinding.inflate(getLayoutInflater());
+        binding.getRoot().setOnTouchListener(new LeftSwipeListener(getContext(), ((MainActivity)getActivity()).getProfileLayoutBinding()));
```

Comment:
Lets check if the getActivity is instance of MainActivity, otherwise it can crash app in future.

Reply:
It seems that we are using `binding.getRoot().setOnTouchListener(new LeftSwipeListener(getContext(), ((MainActivity)getActivity()).getProfileLayoutBinding()));` couple of places can we create ```BaseFragment``` and use extend for it ?